crates/tabby/tests/goldentests.rs

```diff
@@ -84,29 +86,50 @@ async fn golden_test(body: serde_json::Value) -> serde_json::Value {
     actual
 }
 
-async fn assert_golden(body: serde_json::Value) {
-    assert_yaml_snapshot!(golden_test(body).await, {
-        ".id" => "test-id"
-    });
+macro_rules! assert_golden {
+    ($expr:expr) => {
+        insta::assert_yaml_snapshot!(golden_test($expr).await, {
+            ".id" => "test-id"
+        });
+    }
 }
 
+#[cfg(all(target_os = "macos", target_arch = "aarch64"))]
 #[tokio::test]
 async fn run_golden_tests() {
-    wait_for_server().await;
+    wait_for_server(Some("metal")).await;
 
-    assert_golden(json!({
+    assert_golden!(json!({
             "language": "python",
             "segments": {
                 "prefix": "def fib(n):\n    ",
                 "suffix": "\n        return fib(n - 1) + fib(n - 2)"
             }
-    }))
-    .await;
+    }));
 
-    assert_golden(json!({
+    assert_golden!(json!({
             "language": "python",
             "segments": {
                 "prefix": "import datetime\n\ndef parse_expenses(expenses_string):\n    \"\"\"Parse the list of expenses and return the list of triples (date, value, currency).\n    Ignore lines starting with #.\n    Parse the date using datetime.\n    Example expenses_string:\n        2016-01-02 -34.01 USD\n        2016-01-03 2.59 DKK\n        2016-01-03 -2.72 EUR\n    \"\"\"\n    for line in expenses_string.split('\\n'):\n        "
             }
-    })).await;
+    }));
+}
+
+#[tokio::test]
+async fn run_golden_tests_cpu() {
+    wait_for_server(None).await;
```

Comment:
let's specify this explicitly to avoid future change of default flag (e.g we might set metla to default on mac m1/m2)

Reply:
This is really just specifying what to pass for the `--device` flag. The default behavior is to not pass the flag at all, and I think that makes sense. Are you saying you'd rather make a new enum, like `Device::CPU|Cuda|Rocm|Metal`?

Reply:
Oh, so `--device cpu` works too? Wasn't sure; I'll pass that.